torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -353,6 +353,109 @@ std::pair<std::vector<char>, std::vector<at::Tensor>> wireDeserialize(
   return {std::move(payload), std::move(tensors)};
 }
 
+TensorPipeEntry tensorpipeSerialize(const Message& rpcMessage) {
+  tensorpipe::Message tpMessage;
+  std::vector<torch::Tensor> reservedTensors;
+  std::vector<std::vector<uint8_t>> copiedTensors;
+
+  const std::vector<char>& payload = rpcMessage.payload();
+  c10::List<at::Tensor> tensors = cloneSparseTensors(rpcMessage.tensors());
+
+  // Payload
+  tpMessage.data = (uint8_t*)(payload.data());
+  tpMessage.length = payload.size();
+
+  // Metadata - encode rpc message type and message id into
+  // 8 bytes respectively
+  tpMessage.metadata.resize(2 * sizeof(int64_t));
+  int64_t mtype = static_cast<int>(rpcMessage.type());
+  int64_t mId = rpcMessage.id();
+  memcpy(tpMessage.metadata.data(), &mtype, sizeof(int64_t));
+  memcpy(tpMessage.metadata.data() + sizeof(int64_t), &mId, sizeof(int64_t));
```

Comment:
The `data()` method of `std::string` returns a `const char*`. Would `tpMessage.metadata.append(reinterpret_cast<unsigned char*>(&mtype), sizeof(mtype));` work instead?
Also, nit: inconsistent capitalization (`mtype` vs `mId`).